src/modules/sitemap/services/sitemap.service.ts

```diff
@@ -0,0 +1,207 @@
+
+		// Generate general xml
+		const renderedGeneralXml = `<?xml version="1.0" encoding="UTF-8"?>
+		<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+			${generalPages.map(this.renderPage).join('\n')}
+		</urlset>
+		`;
+		xmlUrls.push(await this.uploadXml(renderedGeneralXml, 'general'));
+
+		// Generate itemDetail xml
+		await Promise.all(
+			itemDetailPagesArray.map(async (pages: SitemapItemInfo[], index) => {
+				const renderedXml = `<?xml version="1.0" encoding="UTF-8"?>
+				<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+					${pages.map(this.renderPage).join('\n')}
+				</urlset>
+				`;
+				xmlUrls.push(await this.uploadXml(renderedXml, `itemDetail${index}`));
+			})
+		);
+
+		// Generate index xml
+		const indexXml = `<?xml version="1.0" encoding="UTF-8"?>
+		<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+			${xmlUrls.map(this.renderIndexPage).join('\n')}
+		</sitemapindex>
+		`;
+		await this.uploadXml(indexXml, `index`);
+	}
+
+	public async getSitemapConfig(): Promise<SitemapConfig> {
+		try {
+			const { app_config: config } = await this.dataService.execute<GetSitemapConfigQuery>(
+				GetSitemapConfigDocument
+			);
+
+			return config[0];
+		} catch (err) {
+			throw new InternalServerErrorException('Failed getting sitemap config', err);
+		}
+	}
+
+	public async getContentPagesPaths(): Promise<string[]> {
+		try {
+			const contentPages = await this.contentPagesService.fetchContentPages(
+				0,
+				50000,
+				'title',
+				'asc',
+				'',
+				{ content_type: { _eq: 'PAGINA' } }
+			);
+			const paths = contentPages[0].map((cp) => cp?.path);
+			return paths;
+		} catch (err) {
+			throw new InternalServerErrorException('Failed getting all the content pages', err);
+		}
+	}
+
+	// Helpers
+	// ------------------------------------------------------------------------
+	private renderPage(pageInfo: SitemapItemInfo): string {
+		return `<url>
+			<loc>${pageInfo.loc}</loc>
+			${pageInfo.lastmod ? `<lastmod>${pageInfo.lastmod}</lastmod>` : ``}
+			<changefreq>${pageInfo.changefreq}</changefreq>
+			${pageInfo.priority ? `<priority>${pageInfo.priority}</priority>` : ``}
+			</url>`;
+	}
+
+	private renderIndexPage(url: string): string {
+		return `<sitemap>
+			<loc>${url}</loc>
+			</sitemap>`;
+	}
+
+	private async uploadXml(xml: string, name: string): Promise<string> {
+		return await this.assetsService.upload(AssetFileType.SITEMAP, {
+			fieldname: name,
+			originalname: name,
+			encoding: '',
+			mimetype: 'text/xml',
+			size: 0,
+			stream: new Readable(),
+			destination: '',
+			filename: name,
+			path: '',
+			buffer: Buffer.from(xml, 'utf-8'),
+		});
+	}
+
+	private blacklistAndPrioritizePages(
+		pages: SitemapItemInfo[],
+		config: SitemapConfig
+	): SitemapItemInfo[] {
+		const configPaths = config.value.map((c) => c.path);
+
+		return pages.reduce((result, page) => {
+			const path = page.loc.substring(process.env.CLIENT_HOST.length);
```

Comment:
OPTIONAL: it might be easier to append the process.env.CLIENT_HOST to the urls in the renderPage function. So you can compare paths easily until you convert the links to xml.